shell/platform/windows/flutter_window.h

```diff
@@ -0,0 +1,120 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_FLUTTER_WINDOW_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_FLUTTER_WINDOW_H_
+
+#pragma once
+#include <windowsx.h>
+#include <string>
+#include <vector>
+
+#include "flutter/shell/platform/common/cpp/client_wrapper/include/flutter/plugin_registrar.h"
+#include "flutter/shell/platform/embedder/embedder.h"
+#include "flutter/shell/platform/windows/angle_surface_manager.h"
+#include "flutter/shell/platform/windows/public/flutter_windows.h"
+#include "flutter/shell/platform/windows/win32_window.h"
+
+#include "flutter/shell/platform/common/cpp/incoming_message_dispatcher.h"
+#include "flutter/shell/platform/windows/key_event_handler.h"
+#include "flutter/shell/platform/windows/keyboard_hook_handler.h"
+#include "flutter/shell/platform/windows/platform_handler.h"
+#include "flutter/shell/platform/windows/text_input_plugin.h"
+#include "flutter/shell/platform/windows/window_state.h"
+
+namespace flutter {
+
+constexpr int base_dpi = 96;  // the Windows DPI system is based on this
+                              // constant for machines running at 100% scaling.
+
+// A win32 flutter window.  In the future, there will likely be a
+// CoreWindow-based FlutterWindow as well.  At the point may make sense to
+// dependency inject the native window rather than inherit.
+struct FlutterWindow
```

Comment:
I just realized in skimming this again that there are two `FlutterWindow` classes; this one and one in the wrapper. Maybe to avoid confusion, call this (and thus also the file itself) something slightly different for now? Since it's not public, it'll be easy to rename it later.

Reply:
Fixed in next commit